### DISCUSSION TITLE
Looking for ideas: How can we add A/B testing to our stack?

### DISCUSSION THREAD

**Discussion by usirin** [kamp-us/pano#462] [General]
We wanna have an a/b testing setup with our applications, and ideally wanna have it in our `create-kampus-app` boilerplate tool.

What are some ways we can do this?

- Self hosted?
- SaaS? (sponsorship deal opportunity)

**Comment by Ketcap** [comment]
Using edge via Vercel, https://vercel.com/blog/edge-config-and-launch-darkly

**Reply by Ketcap** [reply]
Another resource can be used https://devcycle.com/ , https://www.learnwithjason.dev/blog/feature-flags-react-devcycle

**Reply by usirin** [reply]
Just want to make sure we are all on the same page: Feature flags are not necessarily same with A/B testing. 

I've checked both of those, even though they market themselves mainly as "feature flag" companies, they have A/B testing feature as well so we are good with these.

P.S this note is mainly for the future reader: if you confuse A/B testing with feature flags, they are pretty similar and they can be used to do the same things, but they actually have different purposes: feature flags are mainly booleans that controls a release, and can be configured to do a gradual rollout. whereas A/B testing is done to test different approaches to the same problem and run product experiments with at least 2 different variants.

**Comment by EmirBoyaci** [comment]
[PostHog](https://posthog.com/) -> advantage: can be self-hosted | disadvantage: a/b testing is not included in free tier for cloud version

**Comment by EmirBoyaci** [comment]
[FlagSmith](https://flagsmith.com/) -> advantage: free tier can be used for a while | disadvantage: free tier is only for one user

**Comment by EmirBoyaci** [comment]
[GrowthBook](https://www.growthbook.io/) -> advantage: free tier can be used up to 3 users | disadvantage: visual a/b test editor is not included in free tier

**Comment by EmirBoyaci** [comment]
[Unleash](https://www.getunleash.io/plans) -> advantage: self-hosted version includes almost everything we want | disadvantage: only one project
